### PR TITLE
Add `reformat` tweak

### PR DESCRIPTION
**Pull request by mosuem**
To be used when `dart format` had any changes in a stable SDK release.

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: pkgs/blast_repo/lib/src/tweaks/reformat_tweak.dart [hunk 1]
```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
```
Comment (devoncarew) [inline pkgs/blast_repo/lib/src/tweaks/reformat_tweak.dart:1]: nit: copyright date

File: pkgs/blast_repo/lib/src/tweaks/reformat_tweak.dart [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import 'dart:io';
+
+import 'package:firehose/firehose.dart';
+
+import '../repo_tweak.dart';
+import '../utils.dart';
+
+final _instance = ReformatTweak._();
+
+/// Regenerate the latest configuration files for package:mono_repo.
```
Comment (devoncarew) [inline pkgs/blast_repo/lib/src/tweaks/reformat_tweak.dart:15]: These docs need to be updated -

File: pkgs/blast_repo/lib/src/tweaks/reformat_tweak.dart [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    final fixes = <String>[];
+    for (var package in packages) {
+      var exitCode = await runProc(
+        'running dart format for ${package.name}',
+        Platform.resolvedExecutable,
+        ['format', '.', '--set-exit-if-changed'],
+        workingDirectory: package.directory.path,
+        throwOnFailure: false,
+      );
+      if (exitCode == 1) {
```
Comment (devoncarew) [inline pkgs/blast_repo/lib/src/tweaks/reformat_tweak.dart:42]: 👍  - we're using the exit code not as a failure, but as an indication that formatting changes were made

File: pkgs/blast_repo/lib/src/tweaks/reformat_tweak.dart [hunk 4]
```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:firehose/firehose.dart';
```
Comment (devoncarew) [inline pkgs/blast_repo/lib/src/tweaks/reformat_tweak.dart:8]: What is blast_repo using from firehose? Is it something we can in-line w/o a lot of cost?

File: pkgs/blast_repo/pubspec.yaml [hunk 5]
```diff
@@ -9,6 +9,8 @@ environment:
 dependencies:
   args: ^2.3.1
   collection: ^1.17.0
+  firehose:
```
Comment (devoncarew) [inline pkgs/blast_repo/pubspec.yaml:12]: I would avoid this path-relative dep if possible.

**Review by devoncarew — APPROVED** [review]

File: pkgs/blast_repo/lib/src/tweaks/reformat_tweak.dart (hunk 4 above)
Comment (mosuem) [reply pkgs/blast_repo/lib/src/tweaks/reformat_tweak.dart:8]: The `Repository` class - something that is needed in both.

File: pkgs/blast_repo/pubspec.yaml (hunk 5 above)
Comment (mosuem) [reply pkgs/blast_repo/pubspec.yaml:12]: I thought that `blast_repo` not being published means we can relax the principle of reducing the number of imports :) I am also fine with code duplication if that's the lesser evil.

Comment (devoncarew) [reply pkgs/blast_repo/pubspec.yaml:12]: It's not wrong - avoiding publishing another package is more important. Perhaps add a line of docs here - `# Re-using the repo scanning logic from firehose` or similar.

Comment (mosuem) [reply pkgs/blast_repo/pubspec.yaml:12]: I am not sure I understand - does this mean you would rather want to copy the code or add the dep with a comment?

**Review by munificent — APPROVED** [review]

**Pull request closed without merging**